src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -338,70 +84,45 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
   const transaction = useAppSelector(gs => selectTxById(gs, ongoingTxId ?? ''))
 
   const handleOngoingTxIdChange = useCallback(
-    async (txid: string) => {
+    async (txid: string, contractAddress?: string) => {
       if (!accountAddress) return
       setOngoingTxId(serializeTxIndex(accountSpecifier, txid, accountAddress))
+      if (contractAddress) setOngoingTxContractAddress(contractAddress)
     },
     [accountSpecifier, accountAddress],
   )
 
   useEffect(() => {
-    if (transaction && transaction.status !== TxStatus.Pending) {
+    if (accountAddress && transaction && transaction.status !== TxStatus.Pending) {
       if (transaction.status === TxStatus.Confirmed) {
-        fetchFarmingOpportunities()
-        fetchLpOpportunity()
+        moduleLogger.info('Refetching fox lp/farming opportunities')
+        reduxDispatch(
+          foxEthApi.endpoints.getFoxEthLpWalletData.initiate(
```

Comment:
Use RTK query hooks instead, you can destructure `refetch` from the `useGetFoxEthLpWalletDataQuery` hook to achieve the same thing